csFastFloat/FastDoubleParser.cs

```diff
@@ -52,52 +67,296 @@ public  static double FastPath(ParsedNumberString pns)
     }
 
 
-    public static unsafe double ParseDouble(string s, NumberStyles expectedFormat = NumberStyles.Float, char decimal_separator = '.')
+    /// <summary>
+    /// Try parsing a double from a UTF-16 encoded string in the given number style
+    /// </summary>
+    /// <param name="s">input string</param>
+    /// <param name="result">output double value</param>
+    /// <param name="styles">allowed styles for the input string</param>
+    /// <param name="decimal_separator">decimal separator to be used</param>
+    /// <returns>bool : true is sucessfuly parsed</returns>
+    public static unsafe bool TryParseDouble(string s, out double result, NumberStyles styles = NumberStyles.Float, char decimal_separator = '.')
+      => TryParseDouble(s, out _, out result, styles, decimal_separator);
+
+    /// <summary>
+    /// Try parsing a double from a UTF-16 encoded string in the given number style, counting number of consummed caracters
+    /// </summary>
+    /// <param name="s">input string</param>
+    /// <param name="characters_consumed">number of consummed caracters while parsing </param>
+    /// <param name="result">output double value</param>
+    /// <param name="styles">allowed styles for the input string</param>
+    /// <param name="decimal_separator">decimal separator to be used</param>
+    /// <returns>bool : true is sucessfuly parsed</returns>
+    /// <returns></returns>
+    public static unsafe bool TryParseDouble(string s, out int characters_consumed, out double result, NumberStyles styles = NumberStyles.Float, char decimal_separator = '.')
     {
-      if (s == null)
-        ThrowArgumentNull();
-        static void ThrowArgumentNull() => throw new ArgumentNullException(nameof(s));
+      if (string.IsNullOrEmpty(s))
```

Comment:
This check special cases the empty input, that (now) is done for each call to `TryParseDouble` so it will regress perf for the non-empty input case which is more common.

I'd omit this check here. In the case of empty input it's handled later in `TryParseNumber`.
So keep the common case with non-empty input fast, but the "penalty" for the rare case of empty input.